services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteSpec.groovy

```diff
@@ -420,6 +420,63 @@ class AutoRerouteSpec extends HealthCheckSpecification {
         flowHelper.deleteFlow(flow.id)
     }
 
+    def "System doesn't reroute flow to a path with not enough bandwidth available"() {
+        given: "A flow with diverse path available"
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            it.paths.unique(false) { a, b -> a.intersect(b) == [] ? 1 : 0 }.size() >= 2
+        } ?: assumeTrue("No suiting switches found", false)
+
+        def flow = flowHelper.randomFlow(switchPair)
+        flowHelper.addFlow(flow)
+
+        def currentPath = pathHelper.convert(northbound.getFlowPath(flow.id))
+        def altPath = switchPair.paths.find { it != currentPath }
+        List<PathNode> broughtDownPorts = []
+        switchPair.paths.findAll { it != currentPath && it != altPath && it.first().portNo != altPath.first().portNo }
+                .unique { it.first() }
+                .each { path ->
+                    def src = path.first()
+                    broughtDownPorts.add(src)
+                    antiflap.portDown(src.switchId, src.portNo)
+                }
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        when: "Make alt path ISLs to have not enough bandwidth to handle the flow"
+        def altIsls = pathHelper.getInvolvedIsls(altPath)
+        altIsls.each {
+            database.updateIslAvailableBandwidth(it, flow.maximumBandwidth - 1)
+        }
+
+        and: "Break isl on the main path(bring port down on the source switch) to init auto reroute"
+        def islToBreak = pathHelper.getInvolvedIsls(currentPath).first()
+        antiflap.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        Wrappers.wait(antiflapMin + 2) {
+            islUtils.getIslInfo(islToBreak).get().state == IslChangeType.FAILED
+        }
+
+        then: "Flow is not rerouted"
+        pathHelper.convert(northbound.getFlowPath(flow.id)) == currentPath
```

Comment:
I think this verification happens too early. The flow may be in progress of rerouting or sth. I believe it's better to wait for some flow status (DOWN in our case) and only then verify the path state